### PR TITLE
WT-10309 Replace mutex with shared_mutex for dynamic tables in workgen

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

File: bench/workgen/workgen.cpp [hunk 1]
```diff
@@ -512,6 +516,8 @@ WorkloadRunner::start_tables_drop(WT_CONNECTION *conn)
 
                 VERBOSE(*_workload, "Marking pending removal for: " << it->first);
                 icontext->_dyn_table_runtime[it->second]._pending_delete = true;
+                ASSERT(std::find(pending_delete.begin(), pending_delete.end(), it->first) ==
```
Comment (sulabhM) [inline bench/workgen/workgen.cpp:519]: @etienneptl I added this assert as requested in the previous review.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Review by tammybailey — APPROVED** [review]
LGTM! Thanks for getting this work done!

**Comment by sulabhM** [comment]
evergreen merge

**Pull request merged**